smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,676 @@
+from watchdog.events import FileCreatedEvent, FileModifiedEvent
+
+from smartsim._core.control.job import Job, JobEntity
+from smartsim._core.control.jobmanager import JobManager
+from smartsim._core.launcher.stepInfo import StepInfo
+from smartsim._core.launcher.cobalt.cobaltLauncher import CobaltLauncher
+
+from smartsim._core.launcher.launcher import Launcher
+from smartsim._core.launcher.local.local import LocalLauncher
+from smartsim._core.launcher.lsf.lsfLauncher import LSFLauncher
+from smartsim._core.launcher.pbs.pbsLauncher import PBSLauncher
+from smartsim._core.launcher.slurm.slurmLauncher import SlurmLauncher
+from smartsim._core.utils.helpers import get_ts
+from smartsim._core.utils.serialize import TELMON_SUBDIR, MANIFEST_FILENAME
+
+from smartsim.error.errors import SmartSimError
+from smartsim.status import TERMINAL_STATUSES
+
+
+"""
+Telemetry Monitor entrypoint
+"""
+
+# kill is not catchable
+SIGNALS = [signal.SIGINT, signal.SIGQUIT, signal.SIGTERM, signal.SIGABRT]
+_EventClass = t.Literal["start", "stop", "timestep"]
+_ManifestKey = t.Literal["timestamp", "model", "orchestrator", "ensemble", "run_id"]
+_JobKey = t.Tuple[str, str]
+
+
+@dataclass
+class Run:
+    """Model containing entities of an individual start call for an experiment"""
+
+    timestamp: int
+    models: t.List[JobEntity]
+    orchestrators: t.List[JobEntity]
+    ensembles: t.List[JobEntity]
+
+    def flatten(
+        self, filter_fn: t.Optional[t.Callable[[JobEntity], bool]] = None
+    ) -> t.List[JobEntity]:
+        """Flatten runs into a list of SmartSimEntity run events"""
+        entities = self.models + self.orchestrators + self.ensembles
+        if filter_fn:
+            entities = [entity for entity in entities if filter_fn(entity)]
+        return entities
+
+
+@dataclass
+class RuntimeManifest:
+    """The runtime manifest holds meta information about the experiment entities created
+    at runtime to satisfy the experiment requirements."""
+
+    name: str
+    path: pathlib.Path
+    launcher: str
+    runs: t.List[Run] = field(default_factory=list)
+
+
+def hydrate_persistable(
+    entity_type: str,
+    persistable_entity: t.Dict[str, t.Any],
+    exp_dir: pathlib.Path,
+) -> t.List[JobEntity]:
+    """Map entity data persisted in a manifest file to an object"""
+    entities: t.List[JobEntity] = []
+
+    if "shards" in persistable_entity or "models" in persistable_entity:
+        container = "shards" if "shards" in persistable_entity else "models"
+        for shard in persistable_entity[container]:
+            entity = JobEntity()
+            metadata = shard["telemetry_metadata"]
+            status_dir = pathlib.Path(metadata.get("status_dir"))
+
+            entity.type = entity_type
+            entity.name = persistable_entity["name"]
+            entity.job_id = str(metadata.get("job_id", ""))
+            entity.step_id = str(metadata.get("step_id", ""))
```

Comment:
I'm inspecting the parents for ensembles & orchestrators and creating a persistable for the child entity. I updated the set var name to more accurately reflect that. See body of `hydrate_persistables`:

```
# an entity w/parent key creates persistables for entities it contains
parent_keys = {"shards", "models"}
parent_keys = parent_keys.intersection(persistable_entity.keys())
```

Reply:
- I updated the keys being read but didn't update property names - the `Job` class uses `job_id` in place of `task_id`. For now, I'd rather keep the two matching just to avoid more confusion.
- Added some cleanup to replace "None" w/empty string